File: src/main/java/tk/hiddenname/smarthome/exception/SignalTypeNotFoundException.java
package tk.hiddenname.smarthome.exception;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;

@ResponseStatus(value = HttpStatus.NOT_FOUND)
public class SignalTypeNotFoundException extends RuntimeException {

    public SignalTypeNotFoundException(String type) {
        super(String.format("The type '%s' is invalid!", type));
    }
}

File: src/main/java/tk/hiddenname/smarthome/exception/SignalTypeNotFoundException.kt
package tk.hiddenname.smarthome.exception;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;

@ResponseStatus(value = HttpStatus.UNPROCESSABLE_ENTITY)
public class SignalTypeNotFoundException extends RuntimeException {

    public SignalTypeNotFoundException(String type) {
        super(String.format("The type '%s' is invalid!", type));
    }
}
